From 2badec2441d95d693d7125d87fa8867f24f1ea9d Mon Sep 17 00:00:00 2001
From: Markus Armbruster <armbru@redhat.com>
Date: Sat, 2 Nov 2013 15:05:06 +0100
Subject: [PATCH 49/81] rhel: Drop "isa-vga" device

RH-Author: Markus Armbruster <armbru@redhat.com>
Message-id: <1383404707-6015-2-git-send-email-armbru@redhat.com>
Patchwork-id: 55272
O-Subject: [PATCH 7.0 qemu-kvm 1/2] rhel: Drop "isa-vga" device
Bugzilla: 1001088
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>
RH-Acked-by: Marcel Apfelbaum <marcel.a@redhat.com>
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>

From: Markus Armbruster <armbru@redhat.com>

We can drop a device from the build only when it isn't referenced from
code we build.  References come in two forms: extern symbols, and qdev
driver names.

Unresolved extern symbols pose no real danger, because they fail
safely at link time.  The dropped code doesn't define any extern
symbols anyway.

Unresolved driver names get detected only at runtime, so we better be
extra careful.  Occurences of the driver name outside the code dropped
by this patch:

* default_list[] in vl.c: Used only to disable default VGA with
  -device isa-vga.  Safe.

* vga_available() in vl.c: Tests whether the device is compiled in.
  No change, because device "VGA" is still compiled in, therfore safe.

* isa_vga_init() in hw/isa/isa-bus.c: Its only potential caller
  pc_vga_init() never calls in RHEL, because it never gets a null
  pci_bus argument.  Safe.

Signed-off-by: Markus Armbruster <armbru@redhat.com>
---
 default-configs/x86_64-softmmu.mak | 1 -
 1 file changed, 1 deletion(-)

Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 default-configs/x86_64-softmmu.mak |    1 -
 1 files changed, 0 insertions(+), 1 deletions(-)

diff --git a/default-configs/x86_64-softmmu.mak b/default-configs/x86_64-softmmu.mak
index 70d8bd9..e7c48ec 100644
--- a/default-configs/x86_64-softmmu.mak
+++ b/default-configs/x86_64-softmmu.mak
@@ -6,7 +6,6 @@ include usb.mak
 CONFIG_VGA=y
 CONFIG_QXL=$(CONFIG_SPICE)
 CONFIG_VGA_PCI=y
-CONFIG_VGA_ISA=y
 CONFIG_VGA_CIRRUS=y
 CONFIG_VMMOUSE=y
 CONFIG_SERIAL=y
-- 
1.7.1

